includes/PathFilter.h

```diff
@@ -0,0 +1,23 @@
+#ifndef FILTER_H
+#define FILTER_H
+
+#include <string>
+#include <vector>
+
+class PathFilter {
+    private:
+        std::string mRoot;
+        std::vector<std::string> mRootFilters;
+        std::vector<std::string> mFilters;
+
+    public:
+        PathFilter(const std::string &root);
+        ~PathFilter();
+
+        /** register a glob filter */
+        void addIgnoreGlob(const std::string &glob);
+        /** return true if path should be ignored, false otherwise */
+        bool ignorePath(const std::string &path);
```

Comment:
```suggestion
        bool isIgnored(const std::string &path);
```

I believe the following method can be renamed so it is clearer and more aligned with the `return` type (`bool`).